gunicorn/arbiter.py

```diff
@@ -473,6 +473,8 @@ def manage_workers(self):
         Maintain the number of workers by spawning or killing
         as required.
         """
+        orig_num_workers = self.num_workers
+
         if len(self.WORKERS.keys()) < self.num_workers:
```

Comment:
I think your reasoning is sound but if the flag is localized to this function then it's pretty fail proof.

At the top of the function, just set `count_changed = False`.
In each of the loops you can set `count_changed =  True`.
At the bottom of the function you can check `count_changed`.